### PR TITLE
add Subproc class to handle env change needed for subprocess

### PR DESCRIPTION
**Pull request by tamarrow**
Due to bugs in pyinstaller and setuptools we need to modify the
environment when forking a command.

### CODE REVIEW

**Comment by tamarrow** [comment]
#746

File: dcos/subcommand.py [hunk 1]
```diff
@@ -686,6 +653,69 @@ def package_json(self):
...
+    """Represents a wrapper for subprocess
+    """
+
+    def __init__(self):
+        self._env = os.environ.copy()
+        lib_path = 'LD_LIBRARY_PATH'
+        lib_path_value = self._env.get(lib_path + '_ORIG')
+        # remove pyinstaller overriding `LD_LIBRARY_PATH`
+        # remvoe with stable release of fix:
+        # https://github.com/pyinstaller/pyinstaller/pull/2148
```
Comment (jsancio) [inline dcos/subcommand.py]: Let's file an issue to remove the work around once that is fixed in pyinstaller. What release should we target?

File: dcos/subcommand.py [hunk 2]
```diff
@@ -686,6 +653,69 @@ def package_json(self):
...
+        if lib_path_value is not None:
+            self._env[lib_path] = lib_path_value
+        elif self._env.get(lib_path):
+            del self._env[lib_path]
+
+        # Setuptools overrides path to executable from virtualenv,
+        # modify this so we can specify a different path
```
Comment (jsancio) [inline dcos/subcommand.py]: Did we end up filing an issue against `setuptools` or asking them about the problem?

File: dcos/subcommand.py [hunk 3]
```diff
@@ -686,6 +653,69 @@ def package_json(self):
             return util.load_json(package_json_file)
 
 
+class Subproc():
```
Comment (jsancio) [inline dcos/subcommand.py]: We should move this out of the `subcommand` module. Are we using this for _all_ exec/fork now?

Comment (tamarrow) [inline dcos/subcommand.py]: yes

File: dcos/subcommand.py (hunk 2 above)
Comment (tamarrow) [inline dcos/subcommand.py]: yes https://github.com/pypa/setuptools/issues/769

File: dcos/subcommand.py (hunk 1 above)
Comment (tamarrow) [inline dcos/subcommand.py]: created #749. I targeted `5.0` which should happen with DCOS 1.10 release.

**Pull request merged**